Fix an annoying MSVC warning and remove some ancient stuff

vadz opened the pull request.

vadz committed:
Remove unused code inside MSVC 6 version checks

We don't support MSVC 6 (__VISUALC__ == 1200) or eVC (1201) since more
than 5 years (see f4b80e533752448952cf181011d15e59061e8b9a), so it's
useless to keep code compiled only when this compiler is used.

vadz committed:
Remove obsolete wxUSE_IOSTREAMH option

This is always 0 for any still supported compiler, so remove the option
and configure checks for it.

Still define it as 0 for compatibility, just in case it's used outside
of the library.

vadz committed:
Get rid of spurious warning in CRT malloc.h with MSVS 2017

Recent MSVS 2017 versions (15.9.x) as well as MSVS 2019 give warning
C4548 about "expression before comma having no effect" when including
standard CRT headers.

This happens because when building projects targeting 8.1 SDK, as the
IDE uses the hard coded 10.0.240.0 version of UCRT headers in this case
(see \Common7\IDE\VC\VCTargets\Microsoft.Cpp.Common.props), and the
headers in this version haven't been, and will never be, updated to
avoid this warning, as was done in later 10.0.x UCRT versions.

Fix this by explicitly disabling the warning in wx/beforestd.h, as even
setting the warning level to 1 for the standard headers somehow isn't
enough to suppress it if it's enabled.

catalinr commented on samples/typetest/typetest.cpp:
```diff
@@ -34,11 +34,7 @@
 
 #include "wx/ioswrap.h"
```
Shouldn't this one be removed or replaced, especially in the samples, now that it's obsolete?
It's also present in `samples/docview/doc.cpp` and `samples/widgets/textctrl.cpp`.

vadz replied:
Good point, thanks!

vadz committed:
Stop using wx/ioswrap.h in the samples

This is not necessary any longer after wxUSE_IOSTREAMH removal and it
makes the sample code more clear.
